src/api/app/policies/workflow_run_policy.rb

```diff
@@ -0,0 +1,21 @@
+class WorkflowRunPolicy < ApplicationPolicy
+  class Scope
+    def initialize(user, scope, opts = {})
+      @user = user
+      @scope = scope
+      @opts = opts
+    end
+
+    def resolve
+      token = Token.find_by(id: opts[:token_id])
+      raise Pundit::NotAuthorizedError, 'the token user not authorized to access the workflow runs' unless token.user == user
+      raise Pundit::NotAuthorizedError, 'the token is not of type workflow' unless token.type == 'Token::Workflow'
+
+      scope.where(token_id: token.id)
```

Comment:
Do you want to add `order(created_at: :desc)` or `order(id: :desc)`, so we show the last workflow_run at the top?

Reply:
good point, applied!